### PR TITLE
refactor: cobra.util.util

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] tests added/passed

This PR refactors `cobra.util.util` module and adds additional tests for the module.

### CODE REVIEW

File: src/cobra/util/util.py [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
+    -------
+    str
+        The shortened string.
+
+    """
     if len(string) > max_length:
-        string = string[: max_length - 3]
-        string += "..."
+        string = f"{string[: max_length - 3]}..."
     return string
```
Comment (Midnighter) [inline src/cobra/util/util.py]: Can you replace the body of the function with [textwrap.shorten](https://docs.python.org/3/library/textwrap.html#textwrap.shorten), please?

```py
return textwrap.shorten(string, width=max_length, placeholder="...")
```

Comment (Midnighter) [reply src/cobra/util/util.py]: A future refactoring could then be to replace all calls to `format_long_string` with `textwrap.shorten` directly.

File: src/cobra/util/util.py [hunk 2]
```diff
@@ -1,27 +1,62 @@
...
+
+        Returns
+        -------
+        Any
+            The retrieved object.
 
-    def __getitem__(self, item):
+        """
         try:
             return dict.__getitem__(self, item)
```
Comment (Midnighter) [inline src/cobra/util/util.py]:
```suggestion
            value = super().__getitem__(self, item)
```

File: src/cobra/util/util.py [hunk 3]
```diff
@@ -1,27 +1,62 @@
...
         except KeyError:
             value = self[item] = type(self)()
             return value
```
Comment (Midnighter) [inline src/cobra/util/util.py]:
```suggestion
        return value
```

File: src/cobra/test/test_util/test_util.py [hunk 4]
```diff
@@ -1,9 +1,37 @@
...
+    )
+    expected_output = "This is a really long string, but is it long en..."
+    assert expected_output == format_long_string(long_string)
+
+
+def test_format_long_string_with_short_string() -> None:
+    """Test functionality by formatting a short string."""
+    short_string = "This is short string."
+    assert short_string == format_long_string(short_string)
+
```
Comment (Midnighter) [inline src/cobra/test/test_util/test_util.py]: You could work with `@pytest.mark.parametrize` to condense these two test definitions into one and simply test different instances.

**Review by Midnighter — CHANGES_REQUESTED** [review]
Nice improvements. Some comments but looks good. The single module + test module PRs are much easier to review :smiley:

File: src/cobra/util/util.py (hunk 1 above)
Comment (synchon) [reply src/cobra/util/util.py]: Perfect.

File: src/cobra/test/test_util/test_util.py (hunk 4 above)
Comment (synchon) [reply src/cobra/test/test_util/test_util.py]: Yes I'll update it.

**Review by Midnighter — APPROVED** [review]
Will merge when tests have passed.

**Pull request merged**